### ISSUE TITLE
cannot import KgeModel, torch.fft is missing?

### ISSUE THREAD

**Issue by fratajcz** [uma-pi1/kge#160]
Hi!

I just trained a model with the CLI and now i want to play with it a bit.

Unfortunately, when I import the KgeModel (from kge.model import KgeModel) I get the following Error:

```
Traceback (most recent call last):
  File "/home/fratajcz/benchmark/src/load_libkge.py", line 72, in <module>
    from kge.model import KgeModel
  File "/home/fratajcz/kge/kge/model/__init__.py", line 4, in <module>
    from kge.model.embedder.lookup_embedder import LookupEmbedder
  File "/home/fratajcz/kge/kge/model/embedder/lookup_embedder.py", line 6, in <module>
    from kge.job import Job
  File "/home/fratajcz/kge/kge/job/__init__.py", line 10, in <module>
    from kge.job.ax_search import AxSearchJob
  File "/home/fratajcz/kge/kge/job/ax_search.py", line 3, in <module>
    from ax import Models
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/ax/__init__.py", line 7, in <module>
    from ax.core import (
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/ax/core/__init__.py", line 8, in <module>
    from ax.core.arm import Arm
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/ax/core/arm.py", line 14, in <module>
    from ax.core.types import TParameterization
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/ax/core/types.py", line 11, in <module>
    from botorch.acquisition import AcquisitionFunction
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/__init__.py", line 7, in <module>
    from botorch import (
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/acquisition/__init__.py", line 7, in <module>
    from botorch.acquisition.acquisition import (
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/acquisition/acquisition.py", line 18, in <module>
    from botorch.models.model import Model
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/models/__init__.py", line 7, in <module>
    from botorch.models.cost import AffineFidelityCostModel
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/models/cost.py", line 16, in <module>
    from botorch.models.deterministic import DeterministicModel
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/models/deterministic.py", line 20, in <module>
    from botorch.models.model import Model
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/models/model.py", line 17, in <module>
    from botorch.posteriors import Posterior
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/posteriors/__init__.py", line 8, in <module>
    from botorch.posteriors.gpytorch import GPyTorchPosterior
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/botorch/posteriors/gpytorch.py", line 15, in <module>
    import gpytorch
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/gpytorch/__init__.py", line 2, in <module>
    from . import (
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/gpytorch/distributions/__init__.py", line 3, in <module>
    from .delta import Delta
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/gpytorch/distributions/delta.py", line 9, in <module>
    from .multivariate_normal import MultivariateNormal
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/gpytorch/distributions/multivariate_normal.py", line 12, in <module>
    from ..lazy import DiagLazyTensor, LazyTensor, delazify, lazify
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/gpytorch/lazy/__init__.py", line 28, in <module>
    from .toeplitz_lazy_tensor import ToeplitzLazyTensor
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/gpytorch/lazy/toeplitz_lazy_tensor.py", line 5, in <module>
    from ..utils.toeplitz import sym_toeplitz_derivative_quadratic_form, sym_toeplitz_matmul
  File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/gpytorch/utils/toeplitz.py", line 4, in <module>
    from torch.fft import fft, ifft
ModuleNotFoundError: No module named 'torch.fft'
```

When I try to install pytorch-fft with pip I get an deprecated import error: 
```
pip install pytorch-fft
Collecting pytorch-fft
  Using cached pytorch_fft-0.15.tar.gz (7.8 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/fratajcz/anaconda3/envs/pygeo/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-0sdj9e0k/pytorch-fft/setup.py'"'"'; __file__='"'"'/tmp/pip-install-0sdj9e0k/pytorch-fft/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-rw4i_5tp
         cwd: /tmp/pip-install-0sdj9e0k/pytorch-fft/
    Complete output (9 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-0sdj9e0k/pytorch-fft/setup.py", line 6, in <module>
        import build
      File "/tmp/pip-install-0sdj9e0k/pytorch-fft/build.py", line 3, in <module>
        from torch.utils.ffi import create_extension
      File "/home/fratajcz/anaconda3/envs/pygeo/lib/python3.8/site-packages/torch/utils/ffi/__init__.py", line 1, in <module>
        raise ImportError("torch.utils.ffi is deprecated. Please use cpp extensions instead.")
    ImportError: torch.utils.ffi is deprecated. Please use cpp extensions instead.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

How can i get past this?

I am using Python 3.8.5 with torch version 1.6.0.

**Comment by fratajcz** [comment]
I just found out that the error is coming from "import ax". 

I updated ax-platform to the newest version (0.1.19) which also updated botorch (0.3.3) and pytorch (1.7.0), and the Issue is gone. I dont know if the rest works now, though. In your setup.py you specify ax-platform==0.1.10 which seems to cause the trouble.

**Comment by samuelbroscheit** [comment]
Thank you for reporting. Yes the Ax version is fixed because we have to check the correct behaviour w.r.t. API calls to Ax. But the pytorch and ax version depend on each other but our setup.py allows always the newest pytorch version.

So either we go back to earlier pytorch versions that are compatible with ax==0.1.10 or we update to the versions reported by fratajcz.

**Comment by rgemulla** [comment]
I suggest to go ahead and update ax (or even "unconstrain" it in the hope that the part of the API that we use is now stable). To do that, we'll need to check if this requires changes on our side.

@samuelbroscheit Did you do the last update of the ax version / know what to look at?

**Comment by samuelbroscheit** [comment]
@rgemulla  Yes I can do it.

**Comment by samuelbroscheit** [comment]
The problem with unconstraining both --- torch and ax --- is that ax is not concurrently updated and there usually is something broke in ax with new torch versions because it depdends on gptorch and botorch. There is usually a window of a few weeks within the newest torch and the current ax version do not work with each other. 

We have 30 to 40 clones of the repository per week, which we risk of loosing if the install already fails. Therefore I suggest to constrain them and regularly do an update to newer versions.

**Comment by AdrianKs** [comment]
Yes, I'd also vote to constrain both

**Comment by rufex2001** [comment]
Same!

**Comment by rgemulla** [comment]
Addressed by #163
